src/org-unit-tree/OrgUnitTree.component.js

```diff
@@ -89,19 +89,26 @@ class OrgUnitTree extends React.Component {
             this.setState({ loading: true });
 
             const childrenIds = root.children.map(({ id }) => id);
+            const fields = {
+                id: true,
+                level: true,
+                displayName: true,
+                shortName: true,
+                children: true,
+                path: true,
+                parent: true,
+            };
+
+            if (onChildrenLoaded?.fields) {
+                onChildrenLoaded.fields.forEach(field => {
+                    fields[field] = true;
+                });
+            }
+
```

Comment:
fields is an internal variable, so no big a big deal, but to keep the FP/immutable style, we could refactor it with lodash. Proposal (untests): 

```typescript
            const extraFields = _(onChildrenLoaded.fields || [])
                .map(field => [field, true])
                .fromPairs()
                .value();

            const fields = {
                id: true,
                level: true,
                displayName: true,
                shortName: true,
                children: true,
                path: true,
                parent: true,
                ...extraFields,
            };
```